### ISSUE TITLE
Cubari: generic JSON proxy

### ISSUE THREAD

**Issue by ObserverOfTime** [appu1232/guyamoe#116]
The current gist proxy is limited to GitHub users. It would be desirable to create a generic proxy—much like the FoolSlide one—that accepts arbitrary JSON URL (which adhere to the schema) in order to proxy any anonymous paste service.

**Comment by Algoinde** [comment]
The reason we're not doing sideloading JSONs is because we'd expose the proxy server IP otherwise. There are ways to resolve that, but I'm actually doing a full rewrite of Cubari, and the schema is changed in the new one. Arbitrary JSONs should not be an issue in the new version.

**Comment by ObserverOfTime** [comment]
Is there a rough ETA on said rewrite?

**Comment by Algoinde** [comment]
Can't say for sure. I'm busy surviving now, optimistic estimate would be a month, I think? If that's too long of a wait for what you're trying to do, @funkyhippo could weigh in if it's possible to query JSONs through a CORS proxy and/or add another whitelisted service if needed. What do you plan to use to host the JSONs?

**Comment by Daktyl198** [comment]
I think the general idea is to allow for the creation of (or modification of an existing) self-hosted scanlation CMS that would host the images and information, but would proxy to cubari’s superior reader for actual reading using auto-generated json.

Allowing cubari to read any properly formatted json file supplied to it (especially if you allow api-style urls and not just straight .json files) would allow such a cms to exist without messing around with the GitHub api.

**Comment by Algoinde** [comment]
Yeah, I understand the use-case perfectly. The new version should be a self-sufficient package that you can use on top of whatever you need, not even necessarily making use of the proxy, even. So not sure if you would want to invest in an infrastructure like that at this moment, as it's slowly becoming legacy. If you do, I'll let Funky answer if the arbitrary-source JSONs would be easy to tack on, as he's the main proxy dev.

**Comment by ObserverOfTime** [comment]
Please don't license the self-sufficient package under AGPL, thank you very much.

**Comment by Daktyl198** [comment]
The ability to integrate the cubari reader into a project without linking out to a separate site would be perfect. However, I hope that it maintains a json schema input as it’s about the most project-agnostic way I can think of to “talk” to the reader.

**Comment by Algoinde** [comment]
> Please don't license the self-sufficient package under AGPL, thank you very much.

What are the particular blockers in AGPL?

**Comment by ObserverOfTime** [comment]
Any product or service that depends on AGPL-licensed code, or includes anything copied or derived from AGPL-licensed code, is **required** to be released under the AGPL license itself.

**Comment by Algoinde** [comment]
Ah right, that's a thing. Yeah, we'll consider that for v2. ~~GPLv3 anyone~~

**Comment by Algoinde** [comment]
Either way, we've talked with Daktyl just now, for what the plan is, you could just write another proxy (https://github.com/appu1232/cubarimoe/tree/master/proxy/sources) for the specific CMS and we can merge that. It could be then reused for 2.0, in theory.

**Comment by ObserverOfTime** [comment]
> Yeah, we'll consider that for v2.

Personally, I think MPL-2.0 would be a fitting license.

> you could just write another proxy

I suppose I'll do that for now.

P.S. I wasn't aware the cubarimoe repo was a thing. You can transfer this issue there.
